packages/gestalt/src/Layer.js

```diff
@@ -9,17 +9,24 @@ type Props = {|
 export default class Layer extends React.Component<Props> {
   constructor(props: Props) {
     super(props);
-    this.el = document.createElement('div');
+    if (typeof document !== 'undefined' && document.createElement) {
+      this.el = document.createElement('div');
+    } else {
+      // eslint-disable-next-line no-console
+      console.warn(
+        'Using Layer without document present. Children will be rendered directly.'
+      );
+    }
   }
 
   componentDidMount() {
-    if (document.body) {
+    if (typeof document !== 'undefined' && document.body) {
```

Comment:
I think you can avoid the `document` checks here by just testing for `this.el` (though technically Flow should be able to figure this out).

Reply:
Flow can't figure it out, sadly.
![screen shot 2019-01-09 at 1 39 23 pm](https://user-images.githubusercontent.com/1767822/50930154-12462880-1414-11e9-8bf3-d7c86fbd250c.png)